heroic-component/src/main/java/com/spotify/heroic/tracing/RandomParentSampler.kt

```diff
@@ -0,0 +1,44 @@
+package com.spotify.heroic.tracing
+
+import io.opencensus.trace.*
+import java.util.concurrent.ThreadLocalRandom
+
+/**
+ * The random parent sampler does not use the sampling decision from a parent when the span says
+ * to sample. Instead it randomly samples spans.
+ *
+ * This will cause traces to not be complete and will likely not capture the outliers! It can
+ * still be useful to use this sampler when you have potentially tens of thousands of spans being
+ * created, as this many spans cannot be shown in any meaningful way within the current UX of
+ * many tracing solutions.
+ */
+class RandomParentSampler(private val probability: Double) : Sampler() {
+    init {
+        if (probability < 0.0 || probability > 1.0) {
+            throw IllegalArgumentException("probability must be in range [0.0, 1.0]")
```

Comment:
Might want to put a `@Throws` annotation on the constructor for usage in Java classes.